### ISSUE TITLE
Parking-v0 reward setting whether is reasonable?

### ISSUE THREAD

**Issue by Obnayuf** [Farama-Foundation/HighwayEnv#500]
sorry for disturbing u.
when I read Hindsight Experience Replay paper, i notice that the author says that shaped reward function doesn't work with HER. So i am confused by ur reward setting in parking-v0."-np.power(np.dot(np.abs(achieved_goal - desired_goal), np.array(self.config["reward_weights"])), p)". I know it isn't a shaped reward. But, the reward still has the problem pointed in HER:" (1) There is a huge discrepancy between what we optimize (i.e. a shaped reward function) and the success condition (i.e.:
is the object within some radius from the goal at the end of the episode); (2) Shaped rewards penalize for inappropriate behaviour (e.g. moving the box in a wrong direction) which may hinder exploration. It can cause the agent to learn not to touch the box at all if it can not manipulate it precisely and we noticed such behaviour in some of our experiments." So, i wanna know why not set the reward to a binary value 0,1. thanks for ur time.

**Comment by eleurent** [comment]
Hi,
The reason is historical: I initially introduced the environment with a dense reward, as this allows to get easy policy improvements (we always get some signal / gradient rewarding getting closer to target), such that a simple PPO agent can solve the task.
But this is still a goal-conditioned task, so I thought it would be a good idea to try out HER, and this did improve the convergence speed (because for any failed trajectory, we get a successful trajectory for free).

I agree with the authors than greedily following the gradient of a shaped reward can in principle fall into local maxima (e.g. maybe you reached the correct position but with the wrong heading, and to fix the heading you need to get further away from the goal which degrades instantaneous rewards) more easily than a binary reward, but I have not observed any particular issue for this specific environment.

So TLDR is that I just havent tried using a binary reward yet, but feel free to do so and report your results here, I might incorporate this as an environment config.
